src/vendor/Duo-Web-v2.6.js

```diff
@@ -1,436 +0,0 @@
-                    'attribute on the iframe element.',
-                    'https://www.duosecurity.com/docs/duoweb#3.-show-the-iframe'
-                );
-            }
-        }
-
-        if (!duoSig || !appSig) {
-            parseSigRequest(getDataAttribute(iframe, 'sigRequest'));
-
-            if (!duoSig || !appSig) {
-                throwError(
-                    'No valid signed request is given.  Be sure to give the ' +
-                    '`sig_request` parameter to Duo.init, or use the ' +
-                    '`data-sig-request` attribute on the iframe element.',
-                    'https://www.duosecurity.com/docs/duoweb#3.-show-the-iframe'
-                );
-            }
-        }
-
-        // if postAction/Argument are defaults, see if they are specified
-        // as data attributes on the iframe
-        if (postAction === '') {
-            postAction = getDataAttribute(iframe, 'postAction') || postAction;
-        }
-
-        if (postArgument === 'sig_response') {
-            postArgument = getDataAttribute(iframe, 'postArgument') || postArgument;
-        }
-
-        // point the iframe at Duo
-        iframe.src = [
-            'https://', host, '/frame/web/v1/auth?tx=', duoSig,
-            '&parent=', encodeURIComponent(document.location.href),
-            '&v=2.6'
-        ].join('');
-
-        // listen for the 'message' event
-        onMessage(onReceivedMessage);
-    }
-
-    /**
-     * We received a postMessage from Duo.  POST back to the primary service
-     * with the response token, and any additional user-supplied parameters
-     * given in form#duo_form.
-     */
-    // function doPostBack(response) {
-    //     // create a hidden input to contain the response token
-    //     var input = document.createElement('input');
-    //     input.type = 'hidden';
-    //     input.name = postArgument;
-    //     input.value = response + ':' + appSig;
-    // 
-    //     // user may supply their own form with additional inputs
-    //     var form = document.getElementById('duo_form');
-    // 
-    //     // if the form doesn't exist, create one
-    //     if (!form) {
-    //         form = document.createElement('form');
-    // 
-    //         // insert the new form after the iframe
-    //         iframe.parentElement.insertBefore(form, iframe.nextSibling);
-    //     }
-    // 
-    //     // make sure we are actually posting to the right place
-    //     form.method = 'POST';
-    //     form.action = postAction;
-    // 
-    //     // add the response token input to the form
-    //     form.appendChild(input);
-    // 
-    //     // away we go!
-    //     if (typeof submitCallback === "function") {
-    //         submitCallback.call(null, form);
-    //     } else {
-    //         form.submit();
-    //     }
-    // }
-
-    // OKTA: Overriding their doPostBack logic because we want to submit this
-    // through ajax, not a postBack
-    function doPostBack(response) {
```

Comment:
@haishengwu-okta  I guess we need to have it checked in for this reason.

Reply:
🤦‍♂ 
Sure. Glad you find the issue this time!

I'm thinking another idea is to create a Duo wrapper to override the post back function
```
define(['duo_web_sdk'], function (Duo) {
  const function oktaDoPostBack = ...
  return Object.assign(Duo, {_doPostBack: oktaDoPostBack });
});
```
I think it shall work and how do you think?
The key point is to make future upgrade a little easier and correct.

Reply:
I tired ur suggestion but I guess we would have to comment out and modify the src file because the doPOstBAck is called within the lib with its internal name and not _doPostback


https://github.com/duosecurity/duo_web_sdk/blob/master/index.js#L408